### PR TITLE
Rust & WASM SDK + tools 

### LINKED ISSUE #6: Build library to interact with the contract
Right now this repo just contains the Aiken contract, along with README instructions for how to interact with the contract

We'll need to make a wrapped for this contract using CML or Lucid to have it be easily usable

### PR DESCRIPTION
**Pull request by gostkin**
Fixes #6 

Main changes:
* Fixed bug with the timestamps in the contract
* Contract dependencies & aiken version are updated
* Rust and WASM SDK added
* Added a rust tool to perform lock / unlock / claim using the SDK
* Added an encode / decode app in nodejs to verify `projected_nft_sdk_wasm` works
* CI / misc

### CODE REVIEW

File: sdk/wasm/example_dapp/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "dependencies": {
+    "@assemblyscript/loader": "^0.27.13",
+    "cbor": "^9.0.1",
+    "deno": "^0.1.1",
+    "lucid-cardano": "^0.10.7",
+    "projected-nft-sdk": "file:../pkg"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
```
Comment (SebastienGllmt) [inline sdk/wasm/example_dapp/package.json:20]: Change to MIT

File: sdk/rust/examples/contract_deploy.rs [hunk 2]
```diff
@@ -0,0 +1,1596 @@
...
+            owner: Owner::PKH(match payment_address.payment_cred().unwrap() {
+                StakeCredential::PubKey { hash, .. } => hash.clone(),
+                StakeCredential::Script { .. } => {
+                    return Err(anyhow!("Expected payment address, not script"))
+                }
+            }),
+            status: Status::Locked,
+        },
+        None => {
+            todo!()
```
Comment (SebastienGllmt) [inline sdk/rust/examples/contract_deploy.rs:303]: TODO?

File: sdk/rust/examples/contract_deploy.rs [hunk 3]
```diff
@@ -0,0 +1,1596 @@
...
+    let ttl = ttl_by_posix(now + 100);
+
+    let script = PlutusV2Script::from_bytes(hex::decode(config.lock_on).unwrap()).unwrap();
+    let contract = PlutusScript::PlutusV2(script);
+
+    let lock_on =
+        EnterpriseAddress::new(network, Credential::new_script(contract.hash())).to_address();
+    println!("Lock on contract: {}", lock_on.to_bech32(None).unwrap());
+
+    // let receipt_asset_name = if let Some(nft) = config.receipt_nft.clone() {
```
Comment (SebastienGllmt) [inline sdk/rust/examples/contract_deploy.rs:246]: Do we need this large block of commented code?

File: sdk/rust/examples/contract_deploy.rs (hunk 2 above)
Comment (gostkin) [reply sdk/rust/examples/contract_deploy.rs:303]: this is receipt use case, which we decided we're not spending time on

File: sdk/rust/examples/contract_deploy.rs (hunk 3 above)
Comment (gostkin) [reply sdk/rust/examples/contract_deploy.rs:246]: if we are ever going to use receipt then yes, it will help

**Pull request merged**